tests/testthat/test-krcomp.R

```diff
@@ -0,0 +1,126 @@
+
+  fit <- mmrm(FEV1 ~ ARMCD + ar1(AVISIT | USUBJID), data = fev_data, method = "Kenward-Roger-Linear")
+  res <- df_1d(fit, contrast = c(0, 1))
+  expected <- c(0.96058142305176, 188.46934887972)
+  expect_equal(res$df, expected[2], tolerance = 1e-4)
+  expect_equal(res$se, expected[1], tolerance = 1e-4)
+
+  fit <- mmrm(FEV1 ~ ARMCD + ar1h(AVISIT | USUBJID), data = fev_data, method = "Kenward-Roger")
+  res <- df_1d(fit, contrast = c(0, 1))
+  expected <- c(0.7590316099633, 188.225339095373)
+  expect_equal(res$df, expected[2], tolerance = 1e-4)
+  expect_equal(res$se, expected[1], tolerance = 1e-2)
+
+  fit <- mmrm(FEV1 ~ ARMCD + ar1h(AVISIT | USUBJID), data = fev_data, method = "Kenward-Roger-Linear")
+  res <- df_1d(fit, contrast = c(0, 1))
+  expected <- c(0.75924807546934, 188.225339095373)
+  expect_equal(res$df, expected[2], tolerance = 1e-4)
+  expect_equal(res$se, expected[1], tolerance = 1e-3)
+
+
+  fit <- mmrm(FEV1 ~ ARMCD + cs(AVISIT | USUBJID), data = fev_data, method = "Kenward-Roger")
+  res <- df_1d(fit, contrast = c(0, 1))
+  expected <- c(0.7964696053595, 177.038485931223)
+  expect_equal(res$df, expected[2], tolerance = 1e-4)
+  expect_equal(res$se, expected[1], tolerance = 1e-3)
+
+  fit <- mmrm(FEV1 ~ ARMCD + cs(AVISIT | USUBJID), data = fev_data, method = "Kenward-Roger-Linear")
+  res <- df_1d(fit, contrast = c(0, 1))
+  expected <- c(0.7964696053595, 177.038485931223)
+  expect_equal(res$df, expected[2], tolerance = 1e-4)
+  expect_equal(res$se, expected[1], tolerance = 1e-4)
+
+  fit <- mmrm(FEV1 ~ ARMCD + csh(AVISIT | USUBJID), data = fev_data, method = "Kenward-Roger")
+  res <- df_1d(fit, contrast = c(0, 1))
+  expected <- c(0.67414806011886, 190.737701349941)
+  expect_equal(res$df, expected[2], tolerance = 1e-3)
+  expect_equal(res$se, expected[1], tolerance = 1e-2)
+
+  fit <- mmrm(FEV1 ~ ARMCD + csh(AVISIT | USUBJID), data = fev_data, method = "Kenward-Roger-Linear")
+  res <- df_1d(fit, contrast = c(0, 1))
+  expected <- c(0.67403858183242, 190.737701349941)
+  expect_equal(res$df, expected[2], tolerance = 1e-3)
+  expect_equal(res$se, expected[1], tolerance = 1e-2)
+
+  fit <- mmrm(FEV1 ~ ARMCD + adh(AVISIT | USUBJID), data = fev_data, method = "Kenward-Roger")
+  res <- df_1d(fit, contrast = c(0, 1))
+  expected <- c(0.66172017349971, 162.393385281755)
+  expect_equal(res$df, expected[2], tolerance = 1e-3)
+  expect_equal(res$se, expected[1], tolerance = 1e-2)
+
+  fit <- mmrm(FEV1 ~ ARMCD + adh(AVISIT | USUBJID), data = fev_data, method = "Kenward-Roger-Linear")
+  res <- df_1d(fit, contrast = c(0, 1))
+  expected <- c(0.66158550758897, 162.393385281755)
+  expect_equal(res$df, expected[2], tolerance = 1e-3)
+  expect_equal(res$se, expected[1], tolerance = 1e-3)
+})
+
+# h_df_1d_kr ---
+
+test_that("h_df_1d_kr works as expected in the standard case", {
+  object_mmrm_kr <- get_mmrm_kr()
+  expect_snapshot_value(h_df_1d_kr(object_mmrm_kr, c(0, 1), TRUE), style = "deparse")
```

Comment:
I wonder if it is worth to add a small test helper `expect_snapshot_tolerance()` e.g. that combines the deparse style with a more relaxed tolerance e.g. 1e-4. can simplify maintenance